internal/builders/bazel/build.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -eu
+#
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+mkdir binaries
+
+# Transfer flags and targets to their respective arrays
+IFS=' ' read -r -a BUILD_FLAGS <<< "${FLAGS}"
+IFS=' ' read -r -a BUILD_TARGETS <<< "${TARGETS}"
+
+# Build with respect to entire arrays of flags and targets
+bazel build "${BUILD_FLAGS[@]}" "${BUILD_TARGETS[@]}"
+
+# Using target string, copy artifact to binaries dir
+for CURR_TARGET in "${BUILD_TARGETS[@]}"; do
+  CD_PATH=${CURR_TARGET%:*}
+  CD_PATH=${CD_PATH////}
```

Comment:
Is this right? I think this removes all '/' characters from the target. IIUC, This would be ok for targets like `//foo:bar` but won't work for targets like `//foo/bar:baz`. In that case `CD_PATH` would end up being `foobar:baz` rather than `foo/bar:baz`.

You also might want to consider relative targets that aren't referencing the workspace root. i.e. `foo/bar:baz` How would we handle getting the right path for those targets?

Reply:
No, I am sorry this is bugged. Caught it as well on a more complex directory test. Will fix it and look into relative targets!

Reply:
Fixed parsing bug, will test it further though and look into relative targets!

Reply:
Bazel requires all targets to referenced with respect to repo: 

> Relative labels cannot be used to refer to targets in other packages; the repository identifier and package name must always be specified in this case.

so we do not need to worry getting the right path for those cases

Reply:
So you can definitely do things like `bazel build target:foo` which will be relative to the current directory (even though you are running it in the workspace root).

For example:
```
git clone https://github.com/bazelbuild/examples
cd examples/cpp-tutorial/stage2
bazel build main:hello-world
```

In this case we are calling the `hello-world` target in the `main` directory. So in this case I would expect it to work even though the target we are referring to doesn't start with `//`.

Reply:
I agree with the example you provided. Since users will input their targets in their workflow job which references the Bazel Builder, I will put into the documentation to reference everything from the root, using // syntax. In the builder it clones the repo and the context is from the root, so all the targets will be relative to the root of the repo. If a target was passed into the workflow in form of `main:hello-world`, it would require additional complexity on the builder side to find the relative target or additional constraints on target name uniqueness.

Do you think that there is a big push for being able to reference targets relatively? I think having to reference the targets from the root is fine, and will document that constraint in the readme.

Reply:
Yeah, you can document that constraint. I think it's pretty common though for folks to run things like `bazel build :main` or `bazel build main:main` so you would want to make that really clear in the documentation that it isn't supported.